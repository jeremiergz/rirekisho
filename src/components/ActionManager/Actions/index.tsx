import FlexBox from 'components/primitives/FlexBox';
import useScreenOrientation from 'hooks/useScreenOrientation';
import React, { MouseEvent, useEffect } from 'react';
import ActionButton from './ActionButton';

const Actions: React.FC<ActionsProps> = ({ onMouseEnter, onMouseLeave, onOrientationChange, opened, sections }) => {
  const orientation = useScreenOrientation();
  const isPortrait = orientation === 'portrait' || (typeof screen !== 'undefined' && screen.height >= 616);
  let delayStart = opened ? sections.length * 25 + 25 : 0;
  useEffect(() => {
    window.addEventListener('orientationchange', onOrientationChange);
    return () => {
      window.removeEventListener('orientationchange', onOrientationChange);
    };
  }, []);
  return (
    <FlexBox
      alignItems="center"
      bottom={isPortrait ? 64 : 0}
      flexDirection={isPortrait ? 'column' : 'row'}
      height={isPortrait ? 'auto' : '100%'}
      justifyContent="center"
      left={isPortrait ? 0 : 'unset'}
      marginBottom={isPortrait ? 2 : 0}
      marginRight={isPortrait ? 0 : 2}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
      pointerEvents={opened ? 'all' : 'none'}
      position="absolute"
      right={isPortrait ? 0 : 64}
      width={isPortrait ? '100%' : 'auto'}
    >
      {sections.map(section => (
        <ActionButton
          Icon={section.Icon}
          key={section.title}
          label={section.title}
          marginBottom={isPortrait ? 1 : 0}
          marginRight={isPortrait ? 0 : 1}
          opened={opened}
          openingDelayInMs={opened ? (delayStart -= 25) : (delayStart += 25)}
          to={section.ref}
        />
      ))}
    </FlexBox>
  );
};

Actions.displayName = 'Actions';

export type ActionsProps = {
  onMouseEnter: (e: MouseEvent<HTMLElement>) => void;
  onMouseLeave: (e: MouseEvent<HTMLElement>) => void;
  onOrientationChange: () => void;
  opened: boolean;
  sections: Models.Section[];
};
export default Actions;
