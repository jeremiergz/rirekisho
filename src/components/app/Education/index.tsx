import { graphql, useStaticQuery } from 'gatsby';
import React, { forwardRef } from 'react';
import Layout from '../../common/Layout';
import FlexBox from '../../primitives/FlexBox';
import Text from '../../primitives/Text';
import Degree from './Degree';
import TimelineDot from './TimelineDot';
import TimelineLine from './TimelineLine';

const Education = forwardRef<HTMLDivElement>((_, ref) => {
  const {
    degreeNodes: { nodes: education },
  } = useStaticQuery<GraphQL.EducationData>(graphql`
    query EducationData {
      degreeNodes: allEducationJson {
        nodes {
          city
          countryCode
          issueDate
          name
          school
        }
      }
    }
  `);
  return (
    <Layout.Section ref={ref} title="education">
      <Layout.Content columnDirection="column-reverse">
        {education
          .sort((a, b) => a.issueDate.localeCompare(b.issueDate))
          .map((degree, index) => (
            <FlexBox
              alignItems="center"
              flex={1}
              flexDirection="column"
              key={degree.name}
              marginBottom={{ _: 4, tablet: 0 }}
              position="relative"
            >
              <Text color="secondary" fontSize={16} fontWeight="bold">
                {degree.issueDate.substring(0, 4)}
              </Text>
              <TimelineDot />
              {index !== 0 && (
                <TimelineLine>
                  <line x1="0%" x2="50%" y1="27%" y2="27%" />
                </TimelineLine>
              )}
              {index !== education.length - 1 && (
                <TimelineLine>
                  <line x1="50%" x2="100%" y1="27%" y2="27%" />
                </TimelineLine>
              )}
              <Degree city={degree.city} countryCode={degree.countryCode} name={degree.name} school={degree.school} />
            </FlexBox>
          ))}
      </Layout.Content>
    </Layout.Section>
  );
});

Education.displayName = 'Education';

export default Education;
