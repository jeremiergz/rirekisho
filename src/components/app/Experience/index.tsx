import { graphql, useStaticQuery } from 'gatsby';
import React, { forwardRef } from 'react';
import Layout from '../../common/Layout';
import ExperienceBlock from './ExperienceBlock';
import Grid from '../../primitives/Grid';

const Experience = forwardRef<HTMLDivElement>((_, ref) => {
  const {
    experienceNodes: { nodes: experiences },
  } = useStaticQuery<GraphQL.ExperienceData>(graphql`
    query ExperienceData {
      experienceNodes: allExperiencesJson {
        nodes {
          company
          companySector
          endDate
          img
          projects {
            client
            clientSector
            description
            name
            tasks
            technologies
          }
          startDate
        }
      }
    }
  `);
  return (
    <Layout.Section ref={ref} title="experience">
      <Layout.Content columnDirection="column-reverse">
        <Grid gridColumnGap={{ _: 32, tablet: 64 }} gridTemplateColumns="auto auto" variant="container">
          {experiences
            .sort((a, b) => b.startDate.localeCompare(a.startDate))
            .map(item => (
              <ExperienceBlock key={item.startDate} gridColumn={{ _: 'span 2', laptopS: 'span 1' }} item={item} />
            ))}
        </Grid>
      </Layout.Content>
    </Layout.Section>
  );
});

Experience.displayName = 'Experience';

export default Experience;
