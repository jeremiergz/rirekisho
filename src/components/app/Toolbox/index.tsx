import { graphql, useStaticQuery } from 'gatsby';
import React, { forwardRef } from 'react';
import LabeledIconsBlock from '../../common/LabeledIconsBlock';
import Layout from '../../common/Layout';

const Toolbox = forwardRef<HTMLDivElement>((_, ref) => {
  const {
    toolNodes: { nodes: tools },
  } = useStaticQuery<GraphQL.ToolboxData>(graphql`
    query ToolboxData {
      toolNodes: allToolboxJson {
        nodes {
          img
          link
          name
          sortOrder
        }
      }
    }
  `);
  return (
    <Layout.Section gridColumn={{ _: 'span 2', tablet: 'span 1' }} ref={ref} title="toolbox">
      <Layout.Content alignItems="center" marginBottom={4}>
        <LabeledIconsBlock items={tools} />
      </Layout.Content>
    </Layout.Section>
  );
});

Toolbox.displayName = 'Toolbox';

export default Toolbox;
