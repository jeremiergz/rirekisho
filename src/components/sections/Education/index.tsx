import Layout from 'components/Layout';
import FlexBox from 'components/primitives/FlexBox';
import Text from 'components/primitives/Text';
import { graphql, useStaticQuery } from 'gatsby';
import React, { forwardRef } from 'react';
import Degree from './Degree';
import TimelineDot from './TimelineDot';
import TimelineLine from './TimelineLine';

const Education = forwardRef<HTMLDivElement>((_, ref) => {
  const {
    degreeNodes: { nodes: education },
  } = useStaticQuery<GraphQL.EducationDataQuery>(graphql`
    query EducationData {
      degreeNodes: allEducationJson {
        nodes {
          city
          countryCode
          issueDate
          name
          school
        }
      }
    }
  `);
  return (
    <Layout.Section contentVisibility="auto" ref={ref} title="education">
      <Layout.Content columnDirection="column-reverse">
        {education
          .sort((a, b) => a.issueDate.localeCompare(b.issueDate))
          .map((degree, index) => (
            <FlexBox
              alignItems="center"
              flex={1}
              flexDirection="column"
              key={degree.name}
              marginBottom={{ _: 4, tablet: 0 }}
              position="relative"
            >
              <Text color="secondary" fontSize={{ _: 20, tablet: 16 }} fontWeight="bold">
                {degree.issueDate.substring(0, 4)}
              </Text>
              <TimelineDot />
              {index !== 0 && (
                <TimelineLine>
                  <line x1="0%" x2="50%" y1="27%" y2="27%" />
                </TimelineLine>
              )}
              {index !== education.length - 1 && (
                <TimelineLine>
                  <line x1="50%" x2="100%" y1="27%" y2="27%" />
                </TimelineLine>
              )}
              <Degree city={degree.city} countryCode={degree.countryCode} name={degree.name} school={degree.school} />
            </FlexBox>
          ))}
      </Layout.Content>
    </Layout.Section>
  );
});

Education.displayName = 'Education';

export default Education;
