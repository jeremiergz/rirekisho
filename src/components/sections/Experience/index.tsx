import Layout from 'components/Layout';
import Grid from 'components/primitives/Grid';
import { graphql, useStaticQuery } from 'gatsby';
import React, { forwardRef } from 'react';
import ExperienceBlock from './ExperienceBlock';

const Experience = forwardRef<HTMLDivElement>((_, ref) => {
  const {
    companyImgNodes: { nodes: companyImgs },
    experienceNodes: { nodes: experiences },
  } = useStaticQuery<GraphQL.ExperienceDataQuery>(graphql`
    query ExperienceData {
      companyImgNodes: allImageSharp(filter: { original: { src: { regex: "/company/" } } }) {
        nodes {
          fixed(height: 24, quality: 100) {
            originalName
            ...GatsbyImageSharpFixed_withWebp
          }
        }
      }
      experienceNodes: allExperiencesJson {
        nodes {
          company
          companySector
          endDate
          img
          projects {
            client
            clientSector
            description
            name
            tasks
            technologies
          }
          startDate
        }
      }
    }
  `);
  const companyImgsIndex = companyImgs.reduce((acc, img) => {
    acc[img.fixed.originalName] = img.fixed;
    return acc;
  }, {});
  return (
    <Layout.Section ref={ref} title="experience">
      <Layout.Content columnDirection="column-reverse">
        <Grid gridColumnGap={{ _: 32, tablet: 64 }} gridTemplateColumns="auto auto" variant="container">
          {experiences
            .sort((a, b) => b.startDate.localeCompare(a.startDate))
            .map(item => {
              const img = companyImgsIndex[item.img];
              return (
                <ExperienceBlock
                  companyImg={img}
                  key={item.startDate}
                  gridColumn={{ _: 'span 2', laptopS: 'span 1' }}
                  item={item as Models.Experience}
                />
              );
            })}
        </Grid>
      </Layout.Content>
    </Layout.Section>
  );
});

Experience.displayName = 'Experience';

export default Experience;
