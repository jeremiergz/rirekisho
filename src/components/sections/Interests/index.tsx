import LabeledIconsBlock from '@/components/common/LabeledIconsBlock';
import Layout from '@/components/common/Layout';
import useInterestsData from '@/components/hooks/data/useInterestsData';
import React, { forwardRef } from 'react';

const Interests = forwardRef<HTMLDivElement>(function Interests(_, ref): JSX.Element {
  const {
    interests: { nodes: interest },
  } = useInterestsData();

  return (
    <Layout.Section className="mt-0" halfWidth ref={ref} title="interests">
      <Layout.Content>
        <LabeledIconsBlock items={interest} />
      </Layout.Content>
    </Layout.Section>
  );
});

export default Interests;
