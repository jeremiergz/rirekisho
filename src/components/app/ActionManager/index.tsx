import React, { FunctionComponent, MouseEvent, useCallback, useRef, useState } from 'react';
import Button from '../../common/Button';
import Box from '../../primitives/Box';
import MenuIcon from '../../svg/Menu';
import Actions from './Actions';

const baseSpacing = '16px';
const positionMapping = {
  'bottom-left': {
    bottom: baseSpacing,
    left: baseSpacing,
  },
  'bottom-right': {
    bottom: baseSpacing,
    right: baseSpacing,
  },
  'top-left': {
    top: baseSpacing,
    left: baseSpacing,
  },
  'top-right': {
    top: baseSpacing,
    right: baseSpacing,
  },
};

const ActionManager: FunctionComponent<ActionManagerProps> = ({ sections, position }) => {
  const actionsRef = useRef<HTMLDivElement>();
  const buttonRef = useRef();
  const [hovered, setHovered] = useState(false);
  const [opened, setOpened] = useState(false);
  const handleMouseEnter = useCallback(() => {
    setHovered(true);
  }, []);
  const handleMouseLeave = useCallback((e: MouseEvent<HTMLElement>) => {
    if (e.relatedTarget !== actionsRef.current && e.relatedTarget !== buttonRef.current) {
      setHovered(false);
    }
  }, []);
  const handleOpen = useCallback(() => {
    setOpened(!opened);
  }, [opened]);
  const handleCloseMenu = () => setOpened(false);
  const positioning = positionMapping[position];
  return (
    <Box height={64} position="fixed" print={false} width={64} zIndex={1} {...positioning}>
      <Actions
        sections={sections}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
        onOrientationChange={handleCloseMenu}
        opened={opened}
      />
      <Button
        animation={hovered || opened ? 'none' : 'pulse 1.5s infinite'}
        backgroundColor="secondary"
        borderRadius="50%"
        height="100%"
        onClick={handleOpen}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
        ref={buttonRef}
        variant="cursor-only"
        width="100%"
      >
        <MenuIcon />
      </Button>
    </Box>
  );
};

ActionManager.displayName = 'ActionManager';

type ActionManagerProps = {
  position: 'bottom-left' | 'bottom-right' | 'top-left' | 'top-right';
  sections: Models.Section[];
};
export default ActionManager;
