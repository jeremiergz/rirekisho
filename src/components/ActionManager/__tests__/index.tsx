import Education from 'components/sections/Education';
import Experience from 'components/sections/Experience';
import Interests from 'components/sections/Interests';
import Skills from 'components/sections/Skills';
import Toolbox from 'components/sections/Toolbox';
import ChevronBottomIcon from 'components/svg/ChevronBottom';
import ChevronTopIcon from 'components/svg/ChevronTop';
import EducationIcon from 'components/svg/Education';
import ExperienceIcon from 'components/svg/Experience';
import InterestsIcon from 'components/svg/Interests';
import SkillsIcon from 'components/svg/Skills';
import ToolboxIcon from 'components/svg/Toolbox';
import React, { createRef } from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from 'theme';
import ActionManager from '..';

describe('components/ActionManager Test Suite', () => {
  it('matches snapshot', () => {
    const experiencesRef = createRef<HTMLDivElement>();
    const skillsRef = createRef<HTMLDivElement>();
    const toolboxRef = createRef<HTMLDivElement>();
    const interestsRef = createRef<HTMLDivElement>();
    const educationRef = createRef<HTMLDivElement>();
    const sections: Models.Section[] = [
      { Icon: ChevronTopIcon, ref: 'top', title: 'top' },
      { Component: Experience, Icon: ExperienceIcon, ref: experiencesRef, title: 'experience' },
      { Component: Skills, Icon: SkillsIcon, ref: skillsRef, title: 'skills' },
      { Component: Toolbox, Icon: ToolboxIcon, ref: toolboxRef, title: 'toolbox' },
      { Component: Interests, Icon: InterestsIcon, ref: interestsRef, title: 'interests' },
      { Component: Education, Icon: EducationIcon, ref: educationRef, title: 'education' },
      { Icon: ChevronBottomIcon, ref: 'bottom', title: 'bottom' },
    ];
    const jsx = (
      <ThemeProvider theme={theme}>
        <ActionManager sections={sections} position="bottom-right" />
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
