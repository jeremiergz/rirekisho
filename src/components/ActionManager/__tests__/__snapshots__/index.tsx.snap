// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ActionManager Test Suite matches snapshot 1`] = `
<div
  className="sc-AxjAm fYLXEs"
  height={64}
  width={64}
>
  <div
    className="sc-AxjAm cnZoIa"
    display="flex"
    height="100%"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    pointerEvents="none"
    width="auto"
  >
    <div
      className="sc-AxjAm diTJxH"
      height="48px"
      opacity={0}
      transform="scale3d(0, 0, 0)"
      width="48px"
    >
      <button
        className="sc-AxjAm sc-AxhUy kSxav"
        height="100%"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="100%"
      >
        <svg
          fill="white"
          height={24}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            d="M7.41 18.205l4.59-4.59 4.59 4.59 1.41-1.41-6-6-6 6 1.41 1.41zM6 7.795v-2h12v2H6z"
          />
        </svg>
      </button>
      <div
        className="sc-AxjAm dMgygb"
        cursor="pointer"
        display="flex"
        height="0"
        onClick={[Function]}
        onMouseLeave={[Function]}
        overflow="hidden"
        width="46px"
        writingMode="vertical-rl"
      >
        <div
          className="sc-AxjAm fMjQHc"
          fontSize={18}
          fontWeight="bold"
          transform="rotate3d(0, 0, 1, -180deg)"
        >
          <span
            className="sc-AxiKw ggvGYh"
            color="secondary"
          >
            top
          </span>
          <span
            className="sc-AxiKw jncuiT"
            color="primary"
          >
            
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-AxjAm gltMof"
      height="48px"
      opacity={0}
      transform="scale3d(0, 0, 0)"
      width="48px"
    >
      <button
        className="sc-AxjAm sc-AxhUy kSxav"
        height="100%"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="100%"
      >
        <svg
          fill="white"
          height={24}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
          />
        </svg>
      </button>
      <div
        className="sc-AxjAm dMgygb"
        cursor="pointer"
        display="flex"
        height="0"
        onClick={[Function]}
        onMouseLeave={[Function]}
        overflow="hidden"
        width="46px"
        writingMode="vertical-rl"
      >
        <div
          className="sc-AxjAm fMjQHc"
          fontSize={18}
          fontWeight="bold"
          transform="rotate3d(0, 0, 1, -180deg)"
        >
          <span
            className="sc-AxiKw ggvGYh"
            color="secondary"
          >
            exp
          </span>
          <span
            className="sc-AxiKw jncuiT"
            color="primary"
          >
            erience
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-AxjAm gWuTTP"
      height="48px"
      opacity={0}
      transform="scale3d(0, 0, 0)"
      width="48px"
    >
      <button
        className="sc-AxjAm sc-AxhUy kSxav"
        height="100%"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="100%"
      >
        <svg
          fill="white"
          height={24}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
          />
        </svg>
      </button>
      <div
        className="sc-AxjAm dMgygb"
        cursor="pointer"
        display="flex"
        height="0"
        onClick={[Function]}
        onMouseLeave={[Function]}
        overflow="hidden"
        width="46px"
        writingMode="vertical-rl"
      >
        <div
          className="sc-AxjAm fMjQHc"
          fontSize={18}
          fontWeight="bold"
          transform="rotate3d(0, 0, 1, -180deg)"
        >
          <span
            className="sc-AxiKw ggvGYh"
            color="secondary"
          >
            ski
          </span>
          <span
            className="sc-AxiKw jncuiT"
            color="primary"
          >
            lls
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-AxjAm livsa-D"
      height="48px"
      opacity={0}
      transform="scale3d(0, 0, 0)"
      width="48px"
    >
      <button
        className="sc-AxjAm sc-AxhUy kSxav"
        height="100%"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="100%"
      >
        <svg
          fill="white"
          height={24}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            d="M21 3H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h5v2h8v-2h5c1.1 0 1.99-.9 1.99-2L23 5c0-1.1-.9-2-2-2zm0 14H3V5h18v12zm-2-9H8v2h11V8zm0 4H8v2h11v-2zM7 8H5v2h2V8zm0 4H5v2h2v-2z"
          />
        </svg>
      </button>
      <div
        className="sc-AxjAm dMgygb"
        cursor="pointer"
        display="flex"
        height="0"
        onClick={[Function]}
        onMouseLeave={[Function]}
        overflow="hidden"
        width="46px"
        writingMode="vertical-rl"
      >
        <div
          className="sc-AxjAm fMjQHc"
          fontSize={18}
          fontWeight="bold"
          transform="rotate3d(0, 0, 1, -180deg)"
        >
          <span
            className="sc-AxiKw ggvGYh"
            color="secondary"
          >
            too
          </span>
          <span
            className="sc-AxiKw jncuiT"
            color="primary"
          >
            lbox
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-AxjAm cA-dAxX"
      height="48px"
      opacity={0}
      transform="scale3d(0, 0, 0)"
      width="48px"
    >
      <button
        className="sc-AxjAm sc-AxhUy kSxav"
        height="100%"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="100%"
      >
        <svg
          fill="white"
          height={24}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            d="M15.5 5.5c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zM5 12c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8.5c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5 3.5 1.6 3.5 3.5-1.6 3.5-3.5 3.5zm5.8-10l2.4-2.4.8.8c1.3 1.3 3 2.1 5.1 2.1V9c-1.5 0-2.7-.6-3.6-1.5l-1.9-1.9c-.5-.4-1-.6-1.6-.6s-1.1.2-1.4.6L7.8 8.4c-.4.4-.6.9-.6 1.4 0 .6.2 1.1.6 1.4L11 14v5h2v-6.2l-2.2-2.3zM19 12c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8.5c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5 3.5 1.6 3.5 3.5-1.6 3.5-3.5 3.5z"
          />
        </svg>
      </button>
      <div
        className="sc-AxjAm dMgygb"
        cursor="pointer"
        display="flex"
        height="0"
        onClick={[Function]}
        onMouseLeave={[Function]}
        overflow="hidden"
        width="46px"
        writingMode="vertical-rl"
      >
        <div
          className="sc-AxjAm fMjQHc"
          fontSize={18}
          fontWeight="bold"
          transform="rotate3d(0, 0, 1, -180deg)"
        >
          <span
            className="sc-AxiKw ggvGYh"
            color="secondary"
          >
            int
          </span>
          <span
            className="sc-AxiKw jncuiT"
            color="primary"
          >
            erests
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-AxjAm dTFNZX"
      height="48px"
      opacity={0}
      transform="scale3d(0, 0, 0)"
      width="48px"
    >
      <button
        className="sc-AxjAm sc-AxhUy kSxav"
        height="100%"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="100%"
      >
        <svg
          fill="white"
          height={24}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3L1 9l11 6 9-4.91V17h2V9L12 3z"
          />
        </svg>
      </button>
      <div
        className="sc-AxjAm dMgygb"
        cursor="pointer"
        display="flex"
        height="0"
        onClick={[Function]}
        onMouseLeave={[Function]}
        overflow="hidden"
        width="46px"
        writingMode="vertical-rl"
      >
        <div
          className="sc-AxjAm fMjQHc"
          fontSize={18}
          fontWeight="bold"
          transform="rotate3d(0, 0, 1, -180deg)"
        >
          <span
            className="sc-AxiKw ggvGYh"
            color="secondary"
          >
            edu
          </span>
          <span
            className="sc-AxiKw jncuiT"
            color="primary"
          >
            cation
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-AxjAm lpjgGD"
      height="48px"
      opacity={0}
      transform="scale3d(0, 0, 0)"
      width="48px"
    >
      <button
        className="sc-AxjAm sc-AxhUy kSxav"
        height="100%"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="100%"
      >
        <svg
          fill="white"
          height={24}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            d="M16.59 5.795L12 10.385l-4.59-4.59L6 7.205l6 6 6-6-1.41-1.41zM18 16.205v2H6v-2h12z"
          />
        </svg>
      </button>
      <div
        className="sc-AxjAm dMgygb"
        cursor="pointer"
        display="flex"
        height="0"
        onClick={[Function]}
        onMouseLeave={[Function]}
        overflow="hidden"
        width="46px"
        writingMode="vertical-rl"
      >
        <div
          className="sc-AxjAm fMjQHc"
          fontSize={18}
          fontWeight="bold"
          transform="rotate3d(0, 0, 1, -180deg)"
        >
          <span
            className="sc-AxiKw ggvGYh"
            color="secondary"
          >
            bot
          </span>
          <span
            className="sc-AxiKw jncuiT"
            color="primary"
          >
            tom
          </span>
        </div>
      </div>
    </div>
  </div>
  <button
    className="sc-AxjAm sc-AxhUy glrBJY"
    height="100%"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    width="100%"
  >
    <svg
      fill="white"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
</div>
`;
